src/Evolution/DiscontinuousGalerkin/Actions/BoundaryConditionsImpl.hpp

```diff
@@ -0,0 +1,742 @@
+          typename... EvolvedVariablesTags, typename... PackageDataVolumeTags,
+          typename... BoundaryConditionVolumeTags, typename... PackageFieldTags,
+          typename... BoundaryCorrectionPackagedDataInputTags>
+void apply_boundary_condition_on_face(
+    const gsl::not_null<db::DataBox<DbTagsList>*> box,
+    [[maybe_unused]] const BoundaryCorrection& boundary_correction,
+    const BoundaryCondition& boundary_condition,
+    const Direction<Dim>& direction,
+    [[maybe_unused]] const Variables<tmpl::list<EvolvedVariablesTags...>>&
+        volume_evolved_vars,
+    [[maybe_unused]] const Variables<
+        db::wrap_tags_in<::Tags::Flux, typename System::flux_variables,
+                         tmpl::size_t<Dim>, Frame::Inertial>>& volume_fluxes,
+    [[maybe_unused]] const Variables<
+        db::wrap_tags_in<::Tags::deriv, typename System::gradient_variables,
+                         tmpl::size_t<Dim>, Frame::Inertial>>& partial_derivs,
+    [[maybe_unused]] const Variables<
+        typename System::compute_volume_time_derivative_terms::temporary_tags>&
+        volume_temporaries,
+    [[maybe_unused]] const Variables<
+        detail::get_primitive_vars_tags_from_system<System>>* const
+        volume_primitive_variables,
+    [[maybe_unused]] const ::dg::Formulation dg_formulation,
+    const Mesh<Dim>& volume_mesh, [[maybe_unused]] const Element<Dim>& element,
+    [[maybe_unused]] const ::ElementMap<Dim, Frame::Grid>& logical_to_grid_map,
+    const domain::CoordinateMapBase<Frame::Grid, Frame::Inertial, Dim>&
+        moving_mesh_map,
+    [[maybe_unused]] const double time,
+    [[maybe_unused]] const std::unordered_map<
+        std::string, std::unique_ptr<domain::FunctionsOfTime::FunctionOfTime>>&
+        functions_of_time,
+    const std::optional<tnsr::I<DataVector, Dim>>& volume_mesh_velocity,
+    const InverseJacobian<DataVector, Dim, Frame::Logical, Frame::Inertial>&
+        volume_inverse_jacobian,
+    [[maybe_unused]] const Scalar<DataVector>& volume_det_inv_jacobian,
+    tmpl::list<PackageDataVolumeTags...> /*meta*/,
+    tmpl::list<PackageFieldTags...> /*meta*/,
+    tmpl::list<BoundaryCorrectionPackagedDataInputTags...> /*meta*/,
+    tmpl::list<BoundaryConditionVolumeTags...> /*meta*/) noexcept {
+  using variables_tag = typename System::variables_tag;
+  using variables_tags = typename variables_tag::tags_list;
+  using flux_variables = typename System::flux_variables;
+  using dt_variables_tags = db::wrap_tags_in<::Tags::dt, variables_tags>;
+  using dt_variables_tag = db::add_tag_prefix<::Tags::dt, variables_tag>;
+
+  const Mesh<Dim - 1> face_mesh = volume_mesh.slice_away(direction.dimension());
+  const size_t number_of_points_on_face = face_mesh.number_of_grid_points();
+
+  // We figure out all the tags we need to project from the interior, both for
+  // the boundary condition computation and for the boundary correction (if
+  // used). We do this by:
```

Comment:
ah, "if used" == `if boundary condition is ghost` because outflow and Bjorhus don't use the boundary correction

Reply:
Ah, yes. I think it would help to say this. I'm not sure if it's best to expand the parenthetical to something like "(if used, i.e., if the boundary condition is ghost)", or to add a new sentence like "Note that the boundary correction is only used is the boundary condition is ghost"... feel free to add in whatever sounds best to you :)